qa/listings.py

```diff
@@ -44,20 +45,65 @@ def run_test(self):
             resp = json.loads(r.text)
             raise TestFailure("ListingsTest - FAIL: Listing POST failed. Reason: %s", resp["reason"])
 
-        # one listing POSTed
-        api_url = node["gateway_url"] + "ob/listings"
+        # one listing POSTed and index returning correct data
+        api_url = vendor["gateway_url"] + "ob/listings"
         r = requests.get(api_url)
-        if r.status_code == 200:
-            if len(json.loads(r.text)) == 1:
-                print("ListingsTest - PASS")
-            else:
-                raise TestFailure("ListingsTest - FAIL: One listing should be returned")
-        elif r.status_code == 404:
+        if r.status_code == 404:
             raise TestFailure("ListingsTest - FAIL: Listings get endpoint not found")
-        else:
+        elif r.status_code != 200:
+            resp = json.loads(r.text)
+            raise TestFailure("ListingsTest - FAIL: Listings GET failed. Reason: %s", resp["reason"])
+
+        resp = json.loads(r.text)
+        if len(resp) != 1:
+            raise TestFailure("ListingsTest - FAIL: One listing should be returned")
+
+        listing = resp[0]
+        if listing["acceptedCurrencies"] != ["tbtc"]:
+            raise TestFailure("ListingsTest - FAIL: Listing should have acceptedCurrences")
+
+        # listing show endpoint returning correct data
+        slug = listing["slug"]
+        api_url = vendor["gateway_url"] + "ob/listing/" + slug
+        r = requests.get(api_url)
+        if r.status_code == 404:
+            raise TestFailure("ListingsTest - FAIL: Listings get endpoint not found")
+        elif r.status_code != 200:
             resp = json.loads(r.text)
             raise TestFailure("ListingsTest - FAIL: Listings GET failed. Reason: %s", resp["reason"])
 
+        resp = json.loads(r.text)
+        if resp["listing"]["metadata"]["acceptedCurrencies"] != ["TBTC"]:
+            raise TestFailure("ListingsTest - FAIL: Listing should have acceptedCurrences in metadata")
+
+        # check vendor's index from another node
+        api_url = browser["gateway_url"] + "ob/listings/" + vendor["peerId"]
+        r = requests.get(api_url)
+        if r.status_code == 404:
+            raise TestFailure("ListingsTest - FAIL: Listings get endpoint not found")
+        elif r.status_code != 200:
+            resp = json.loads(r.text)
+            raise TestFailure("ListingsTest - FAIL: Listings GET failed. Reason: %s", resp["reason"])
+        resp = json.loads(r.text)
+        if len(resp) != 1:
+            raise TestFailure("ListingsTest - FAIL: One listing should be returned")
+        if resp[0]["acceptedCurrencies"] != ["tbtc"]:
+            raise TestFailure("ListingsTest - FAIL: Listing should have acceptedCurrences")
```

Comment:
Is there a benefit to checking the listing from another node? And if not, then I'd prefer a unit test in go. (Especially considering these may migrate depending on how we improve our QA, integration test setup?)

Reply:
There is a different code branch taken when getting your own listings versus someone elses so I wanted to test both.

Reply:
I went ahead and made some unit tests anyways.